src/tests/ftest/io/ior_intercept_dfuse_mix.py

```diff
@@ -0,0 +1,239 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import os
+import threading
+import time
+import write_host_file
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand, IorMetrics
+from command_utils import CommandFailure
+
+
+class IorInterceptDfuseMix(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description: Runs IOR only with dfuse and with mix of
+       dfuse and interception library on a single server and multi
+       client settings with basic parameters.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up each test case."""
+        super(IorInterceptDfuseMix, self).setUp()
+        # Following if block can be removed once the constraint
+        # in IorTestBase is removed. # DAOS-3320
+        if self.ior_cmd.api.value == "POSIX":
+            self.hostlist_clients = self.params.get(
+                "test_clients", "/run/hosts/*")
+        self.lock = threading.Lock()
+
+    def test_ior_intercept_dfuse_mix(self):
+        """Jira ID: DAOS-3500.
+
+        Test Description:
+            Purpose of this test is to run ior through dfuse on 4 clients
+            for 5 minutes and capture the metrics and use the
+            intercepiton library by exporting LD_PRELOAD to the libioil.so
+            path on 3 clients and leave 1 client to use dfuse and rerun
+            the above ior and capture the metrics and compare the
+            performance difference and check using interception
+            library make significant performance improvement. Verify the
+            client didn't use the interception library doesn't show any
+            improvement.
+
+        Use case:
+            Run ior with read, write for 5 minutes
+            Run ior with read, write for 5 minutes with interception
+            library
+
+            Compare the results and check whether using interception
+                library provides better performance and not using it
+                does not change the performance.
+
+        :avocado: tags=all,daosio,hw,full_regression,iorinterceptmix
```

Comment:
Is it important to be run on real hardware? I think as we are only concerned with difference in perf numbers for POSIX vs POSIX with interception lib, vms should give the difference in perf numbers too.
Please let me know if I am missing out on some special requirement which needs this test to be run on real hw.

Reply:
These tests in general require more memory that VMs don't offer. This test in specific can run in VM if we try to squeeze the memory requirement but I'd prefer it to run on real hardware because if we have to increase the transfer size, it can utilize both NVMe and SCM.

Reply:
VMs give varying performance depending on system load, where "system" includes things outside the VM.  If we're using performance as a means of determining pass/fail then these tests need to run on real hardware.